### ISSUE TITLE
Crashing if we do not use InstallReferrer library in consumer application

### ISSUE THREAD

**Issue by Prithuld** [snowplow/snowplow-android-tracker#682]
**Describe the bug**
Our app is crashing and it needs InstallReferrer library to be added in our application to get it fixed.

**To Reproduce**
Aleady background is mentioned https://github.com/snowplow/snowplow-android-tracker/issues/647 

**Expected behavior**
We do not want to add any additional library in our application to support snowplow, ideally snowplow should fix it by passing the dependency transitively .
We just want to use snowplow library (no additional one to support it 😞 )

**Screenshots**
Mentioned https://github.com/snowplow/snowplow-android-tracker/issues/647

**Device informatoin (please complete the following information):**
 - Device: [e.g. Pixel XL2]
 - OS: [e.g. Android 9]
 - Browser [e.g. chrome, firefox]
 - Version [e.g. 84]

**Comment by miike** [comment]
@Prithuld Is this the same issue described in #647 or are you reporting a different issue? If so I will close this.

**Comment by Prithuld** [comment]
No,actually we do not prefer to implement InstallReferrer in our application, are you going to provide required dependency transitively?

**Comment by matus-tomlein** [comment]
Hi @Prithuld, thanks for raising this! The tracker does not have a transitive dependency on the InstallReferrer because we wanted to let users who don't wish to track the information provided by the InstallReferrer to use the tracker without any extra dependencies. Tracking the referrer information from the InstallReferrer library is an optional feature that is only enabled in case the dependency is available in the app. Does that make sense?

**Comment by Prithuld** [comment]
ok thanks.. what will be the impact if we do not implement InstallReferrer ? which feature we will loose?

**Comment by matus-tomlein** [comment]
@Prithuld the impact is the `referrer_details` entity that the tracker adds to the application install event based on the information from the InstallReferrer library – [here is the documentation for it](https://docs.snowplow.io/docs/collecting-data/collecting-from-own-applications/mobile-trackers/tracking-events/installation-tracking/#android-only-tracking-referrer-information-from-the-google-play-referrer-library).
